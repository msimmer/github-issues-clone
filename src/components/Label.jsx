// @flow

import * as React from "react";
import { contrastText } from "../util";

const Label = ({
  name,
  color
}: {
  name: string,
  color: string
}): React.Element<"div"> => (
  <div
    className="label"
    style={{ backgroundColor: `#${color}`, color: contrastText(color) }}
  >
    {name}
  </div>
);

export default Label;
