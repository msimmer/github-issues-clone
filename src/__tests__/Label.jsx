import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { configure } from "enzyme";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import Label from "../components/Label";

configure({ adapter: new Adapter() });

describe("<Label />", () => {
  it("contrasts the color", () => {
    const component = shallow(<Label name="foo" color="ffffff" />);
    expect(component.props().style).toHaveProperty("color", "black");
    expect(toJSON(component)).toMatchSnapshot();

    component.setProps({ color: "000000" });
    expect(component.props().style).toHaveProperty("color", "white");
    expect(toJSON(component)).toMatchSnapshot();
  });
});
