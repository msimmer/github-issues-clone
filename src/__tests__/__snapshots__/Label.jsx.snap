// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> contrasts the color 1`] = `
<div
  className="label"
  style={
    Object {
      "backgroundColor": "#ffffff",
      "color": "black",
    }
  }
>
  foo
</div>
`;

exports[`<Label /> contrasts the color 2`] = `
<div
  className="label"
  style={
    Object {
      "backgroundColor": "#000000",
      "color": "white",
    }
  }
>
  foo
</div>
`;
